### DISCUSSION TITLE
WebVTT voice spans

### DISCUSSION THREAD

**Discussion by ryan-lp** [Podcastindex-org/podcast-namespace#594] [Clarification Needed]
Here is a snippet from the commit https://github.com/Podcastindex-org/podcast-namespace/pull/220/commits/4c6a8249f24cf60d302ce35327f05826124bea7b which introduced the WebVTT example into the podcast namespace spec:

```vtt
WEBVTT

00:00:00.179 --> 00:00:02.399
Travis: When you first get
started in podcasting, it's

00:00:02.400 --> 00:00:04.799
almost guaranteed that you're
going to make a handful of rookie
```

Compare this to the first example from the [WebVTT spec](https://www.w3.org/TR/webvtt1/#introduction-caption):

```vtt
WEBVTT

00:11.000 --> 00:13.000
<v Roger Bingham>We are in New York City

00:13.000 --> 00:16.000
<v Roger Bingham>We’re actually at the Lucern Hotel, just down the street
```

Since the WebVTT specification already defines a standard way to provide speaker tags via voice spans, we should have specified this in our standard rather than using plain text such as `Travis: `, which is unreliable to parse out. i.e. Not all words before colons can be safely assumed to be speaker tags. It would have been very nice to have adequate time for feedback before this was adopted.

(It looks like the WebVTT format was proposed and formalised into the spec [in a span of 2 days](https://github.com/Podcastindex-org/podcast-namespace/issues/217). Hmm... See #593)

**Comment by daveajones** [comment]
If the examples are wrong then that needs to be corrected.  During adoption I was assuming people would create VTT files based on the actual VTT spec, not reverse engineering our examples.  The examples are *not* meant to be a replacement for reading the actual format spec.

**Comment by daveajones** [comment]
Example file has been corrected:  https://github.com/Podcastindex-org/podcast-namespace/commit/04e99c5c22a84b1e8a618633818566579abccddd

It appears that someone messed up the example file by using the SRT speaker tag format in it.

**Reply by Dwev** [reply]
I've tried looking for the SRT speaker tag documentation, but it doesn't seem to be part of the official spec.  Is this something that has been added unofficially?

**Comment by nathangathright** [comment]
Do we have any example feeds using VTT correctly? I’d love to test how robust the support is across apps.

**Reply by daveajones** [reply]
I believe Peertube feeds use VTT.

**Reply by jamescridland** [reply]
Podnews has a VTT feed. It works in a browser - see the bottom of https://podnews.net/article/apple-podcasts-transcriptions-faq - and is just the output of whisper.cpp

I cannot (yet?) get any third-party transcript into Apple Podcasts. I've tried mine, Buzzsprout and RSS(dot)com.

**Reply by nathangathright** [reply]
@jamescridland I checked your feed first, but I didn’t spot one that included a speaker label like: `<v James Cridland>`, and I wanted to test how they rendered in Podverse/Fountain/Steno/etc.

**Comment by daveajones** [comment]
The example here has also been fixed:  https://github.com/Podcastindex-org/podcast-namespace/blob/main/transcripts/transcripts.md#webvtt

And, language has been added to let developers know to read the actual format spec instead of relying on eyeballing the examples.

**Reply by ryan-lp** [reply]
Part of my comment was that original changes were slipped into the spec too quickly. Last time in a span of 2 days, but this time in a span of 4 days. Or really, your first reply to my post was to say that you had already fixed it. I see you have not really taken on board that feedback, which was the reason these mistakes get introduced in the first place. There are similar mistakes with the SRT format which I have reported before, I never got a chance to review the original SRT spec, and those mistakes are still not fixed. So there is another issue with the process where critical issues are overlooked because they don't have enough popularity or attention compared to more popular topics, while everybody is busy wanting to add new features to the spec like chat, publisher feeds, and most of all, anything to do with crypto payments which drain a huge amount of attention and focus away from other very important issues that get neglected for years, like issues with the transcript spec which hasn't fundamentally changed in 4 years.

> And, language has been added to let developers know to read the actual format spec instead of relying on eyeballing the examples.

I disagree with the principle of this, the original spec supports far too many features, many of which we should definitely not support for various reasons which I can elaborate on if needed. There are many different ways that VTT can be used in practice, and there are many different ways that SRT can be used in practice, and we should not permit any of this variation because it would make it impossible to build apps to take advantage of a format when we don't know what we're going to extpect. An example with SRT is permitting unlimited line widths, or putting one word per  line, it is our job to standardise that, and not using a link to the original spec as a fallback just in case our own spec was incomplete or inaccurate. Speaking of which,

With the VTT example, you introduced another mistake. The start tag `<v>` is implicitly closed by the end of the cue, so each and every cue needs a voice span, or else it means that there is no voice associated with it. My critique isn't that you made a mistake and you should have known VTT better, my critique is that we need more time to review your changes, and we need a more rigorous process that can be checked before it is merged...

**Reply by jamescridland** [reply]
I have corrected and validated the example (and copy/pasted this into the example file, though I'd question why that's there). It is waiting for @daveajones to review and merge.

The example WebVTT also had an unescaped & sign, which failed [validation](https://w3c.github.io/webvtt.js/parser.html).

**Reply by daveajones** [reply]
Merged.

**Reply by Dwev** [reply]
It is worth including the class data into the voice span examples?  The spec allows for `<v.loud.top John Smith>` for example, which should be able to be parsed, even if it ends up being ignored.
